### PR TITLE
Add AWS Secrets Manager blog post

### PR DESCRIPTION
**Pull request by harrisonmeister**
# About

This is a post about integrating AWS Secrets Manager with Octopus using a new step template.

# Release

N/A

# Blog image idea

I've added a placeholder image for the blog post, using the Azure Key Vault one as an example. Happy for the design to be changed as you and the design team see fit.

# Pre-requisites

- [X] The draft is complete and this post is ready to be reviewed.
- [X] I'm familiar with the [writing for Octopus TL;DR](https://style.octopus.com/writing-at-octopus-tldr).
- [X] The PR build passes validation, and if it doesn't, I've checked the [common validations errors](https://style.octopus.com/writing-at-octopus-tldr#common-validation-errors) and none of those apply.
- [X] The images I've included follow the [rules for working with images](https://style.octopus.com/images#rules-for-screenshots).

## Publication date
You may publish at your discretion :+1:

### CODE REVIEW

File: blog/2021-11/using-aws-secrets-manager-with-octopus/index.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+---
+title: Using AWS Secrets Manager with Octopus
+description: Introducing a new step template to allow secrets stored in AWS Secrets Manager to be used in deployments or runbooks.
+author: mark.harrison@octopus.com
+visibility: public
+published: 2021-11-22-1400
+metaImage: blogimage-usingawssecretsmanagerwithoctopus-2021.png
+bannerImage: blogimage-usingawssecretsmanagerwithoctopus-2021.png
+bannerImageAlt: AWS secrets manager logo and Octopus Deploy logo joined together with a connector
```
Comment (TeganAli) [inline blog/2021-11/using-aws-secrets-manager-with-octopus/index.md:9]:
```suggestion
bannerImageAlt: AWS secrets manager logo and Octopus Deploy logo joined together by a connector
```

File: blog/2021-11/using-aws-secrets-manager-with-octopus/index.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+I've written several step templates that extend the functionality of Octopus to integrate with secrets managers, the last being [Google Cloud Secret Manager](https://octopus.com/blog/using-google-cloud-secret-manager-with-octopus). One major cloud provider I hadn't covered until now was Amazon Web Services (AWS).
+
+In this post, I walk through a new step template, [AWS Secrets Manager - Retrieve Secrets](https://library.octopus.com/step-templates/5d5bd3ae-09a0-41ac-9a45-42a96ee6206a/actiontemplate-aws-secrets-manager-retrieve-secrets), designed to retrieve secrets from AWS Secrets Manager for use in your deployments or runbooks.
+
+:::hint
+You can also read about:
+
+- [Using Azure Key Vault with Octopus](https://octopus.com/blog/using-azure-key-vault-with-octopus)
+- [Using HashiCorp Vault with Octopus Deploy](https://octopus.com/blog/using-hashicorp-vault-with-octopus-deploy)
```
Comment (TeganAli) [inline blog/2021-11/using-aws-secrets-manager-with-octopus/index.md:24]:
```suggestion
- [Using HashiCorp Vault with Octopus](https://octopus.com/blog/using-hashicorp-vault-with-octopus-deploy)
```

File: blog/2021-11/using-aws-secrets-manager-with-octopus/index.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+  **Note:** Multiple secrets can be retrieved by entering each one on a new line. See the [examples](#secret-retrieval-examples) for more information.
+
+- **Print output variable names**: Write out the Octopus [output variable](https://octopus.com/docs/projects/variables/output-variables) names to the task log. Default: `False`.
+
+![Parameters for the step](aws-secrets-manager-retrieve-secrets-step-parameters.png)
+
+### Secret retrieval examples {#secret-retrieval-examples}
+
+Let's use the example of a secret stored in AWS Secrets Manager named **OctoSamples-usercredentials** with two key/value pairs:
```
Comment (TeganAli) [inline blog/2021-11/using-aws-secrets-manager-with-octopus/index.md:112]: @harrisonmeister I updated this sentence as it didn't make sense to me, but just want to check I haven't altered your intended meaning? Thanks :)

Comment (harrisonmeister) [reply blog/2021-11/using-aws-secrets-manager-with-octopus/index.md:112]: No change in the intended meaning 😄. The section is showing the different ways you can enter details of secrets to retrieve their values. The step parameter supports different options and the post offers a chance to give customers a detailed look at examples. But without a specific secret as context the parameter examples might not have made sense - Essentially the sentence was to "set the scene"

File: blog/2021-11/using-aws-secrets-manager-with-octopus/index.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+author: mark.harrison@octopus.com
+visibility: public
+published: 2021-11-22-1400
+metaImage: blogimage-usingawssecretsmanagerwithoctopus-2021.png
+bannerImage: blogimage-usingawssecretsmanagerwithoctopus-2021.png
+bannerImageAlt: AWS secrets manager logo and Octopus Deploy logo joined together by a connector
+tags:
+ - DevOps
+ - Security
+ - Step Templates
```
Comment (TeganAli) [inline blog/2021-11/using-aws-secrets-manager-with-octopus/index.md:13]:
```suggestion
 - Step Templates
 - AWS
```

**Pull request merged**